app/controllers/products_controller.rb

```diff
@@ -0,0 +1,44 @@
+class ProductsController < ApplicationController
+  
+  before_action :current_product, only: [:show, :edit, :update, :destroy]
+  
+  def index
+    @products = Product.all
+  end
+
+  def show
+  end
+
+  def new
+    @product = Product.new
+  end
+
+  def create
+    Product.create(product_params)
+    redirect_to products_path
```

Comment:
what will happen if create will fail? should you maybe display an error message and not redirect?